### PR TITLE
feat(overflow): add overflow_forced_keys envvar

### PR DESCRIPTION
**Pull request by xvello**
- Add the `overflow_forced_keys` config to mirror capture-py's `EVENT_PARTITION_KEYS_TO_OVERRIDE`, to force a key to overflow even if the burst threshold is not reached.
- Add unit test for that feature

`HashSet::contains` has a fast-path for empty state, so I'm just using a `HashSet::new()` by default. There are several hashing algorithms to choose from, but that's not a big concern right now.

### CODE REVIEW

**Review by ellie — APPROVED** [review]

**Comment by ellie** [comment]
one failing test

**Review by tiina303 — APPROVED** [review]

File: capture/src/config.rs [hunk 1]
```diff
@@ -19,6 +19,8 @@ pub struct Config {
     #[envconfig(default = "1000")]
     pub burst_limit: NonZeroU32,
 
+    pub overflow_forced_keys: Option<String>, // Coma-delimited keys
```
Comment (tomasfarias) [inline capture/src/config.rs:22]: nit: I think I'd prefer to have the `HashSet` here, as it feels like we are leaking implementation details to `PartitionLimiter`, who only cares about the `HashSet` of keys at the end.

Not a big deal while this config is only used there, but perhaps something to refactor later if you agree.

**Review by tomasfarias — APPROVED** [review]
lgtm

File: capture/src/config.rs (hunk 1 above)
Comment (xvello) [reply capture/src/config.rs:22]: Agreed! I'll want to replicate [plugin-server's ValueMatchers](https://github.com/PostHog/posthog/blob/984bef92a89223a48ae64c7c106db70a78eafc11/plugin-server/src/config/config.ts#L232-L265), but I'll do that when overhauling the configuration in December

**Pull request merged**